File: chapter-02-document-model/python/insert-basic.py
from pymongo import MongoClient
from datetime import datetime

client = MongoClient("mongodb://bookuser:bookpassword@localhost:10260/?authSource=admin&tls=true&tlsAllowInvalidCertificates=true")

db = client.bookdb
users = db.users

users.insert_one({
    "userId": "u123",
    "name": "Alice",
    "preferences": {
        "language": "en",
        "notifications": True
    },
    "createdAt": datetime.utcnow()
})

print("Document inserted successfully")

File: chapter-02-document-model/nodejs/insert-basic.js
import { MongoClient } from "mongodb";

const uri = "mongodb://localhost:27017";
const client = new MongoClient(uri);

async function run() {
  await client.connect();

  const db = client.db("bookdb");
  const users = db.collection("users");

  await users.insertOne({
    userId: "u123",
    name: "Alice",
    preferences: {
      language: "en",
      notifications: true
    },
    createdAt: new Date()
  });

  console.log("Document inserted successfully");

  await client.close();
}

run().catch(console.error);
